src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,6 +471,38 @@ static int open_device(const char* path, bool try_load_moudle = false)
   return nbd;
 }
 
+static bool find_key_from_conf_file(const char *key)
+{
+  std::vector <std::string> my_sections;
+  g_conf->get_my_sections(my_sections);
+  std::string admin_socket_str;
+  return ((g_conf->get_val_from_conf_file(my_sections, key,
+           admin_socket_str, false) == 0) ? true : false);
```

Comment:
You don't need to add. rbd-nbd parses command line options using global_init(). That means that any config option can be also specified via command line. Just try:
```
rbd-nbd --admin-socket=/tmp/test.asok ${image}
```
and check that after your modifications it still uses /tmp/test.asok in this case.